web/src/main/java/se/inera/intyg/webcert/web/web/controller/testability/facade/util/CreateCertificateTestabilityUtil.java

```diff
@@ -96,6 +108,19 @@ public String createNewCertificate(@NotNull CreateCertificateRequestDTO createCe
             createCertificateRequest.getCertificateTypeVersion()
         );
 
+        if (certificateServiceProfile.active()) {
```

Comment:
If you can combine the active() and the certificateTypeExits() in one private method, you can "flatten" the code depth.

Also after receiving the modelIdDTO you should also check the version returned if it corresponds to the version to create.

Another thing is to make sure you create the version asked for and not the one returned by certificateTypeExists, as it will always return the latest active version. 

If you want you can deal with these things in this PR, or we manage them in the next "iteration" of the testability api, which will be next sprint when we add content with the certificate.

When I write this I think best would be to take it in the next iteration where we can add a testability endpoint in CS to check a specific version.